Feature/adhithya christen/integrate mongo db

adhi0331 opened the pull request:
### Tracking Info

Resolves #6

Make sure your branch name conforms to: `<feature|staging|bugfix|...>/<username>/<3-4 word description separated by dashes>`. Otherwise, please rename your branch and create a new PR.

### Changes

What changes did you make?

We were able to get the app connected to the MongoDB. We also created User models, and routes to get and add users. 

### Testing

How did you confirm your changes worked? 

We tested the routes on postman and were able to get both routes to work properly.

### Confirmation of Change 

Upload a screenshot, if possible. Otherwise, please provide instructions on how to see the change.

Changes can be viewed in the app.ts, models folder, and routes folder.

AmanKAggarwal commented on backend/package.json:
```diff
@@ -14,10 +14,12 @@
   "license": "ISC",
   "dependencies": {
     "dotenv": "^16.0.3",
-    "express": "^4.18.2"
```
Add `body-parser` here. Follow lint instructions for this

AmanKAggarwal commented on backend/src/models/users.ts:
```diff
@@ -0,0 +1,47 @@
+import mongoose from 'mongoose';
+
+interface userInterface{
+    name: String;
```
Use `string` not `String`. Check this out for understanding the difference - https://stackoverflow.com/questions/14727044/what-is-the-difference-between-types-string-and-string

AmanKAggarwal reviewed (changes requested):
Nice work guys!! PR looks very solid but there are few changes to be made before we can merge
- You might have seen the lint errors on the PR. In case you are not familiar, [linting](https://cogut.medium.com/what-is-linting-how-does-a-linter-work-49381f28fc60) is the process of checking the source code for Programmatic as well as Stylistic errors. Often times it can be annoying and time consuming to detect such errors while reviewing PR, which is why we are using ESLint for this. Make sure you read, understand and fix all the errors the lint check revealed to pass it. 
Use the following command to fix linting errors that can be fixed automatically - 
```
npm run lint-fix
```

- You can ignore the warnings about console.log because in the development phase some console logs are very important eg. Connected to Database
- Update body-parser in package.json. Whenever you add a package, make sure the package.json is being updated so that someone could just `npm install` in our repo and start hacking! 
-  GET /user is not yet complete 

Above are the main things I noticed, I also dropped more specific comments in the PR so address those as well. Overall, these are minor changes so great work integrating mongoDB with our server.

AmanKAggarwal commented:
> ### Tracking Info
> Resolves #
> 
> Make sure your branch name conforms to: `<feature|staging|bugfix|...>/<username>/<3-4 word description separated by dashes>`. Otherwise, please rename your branch and create a new PR.
> 
> ### Changes
> What changes did you make?
> 
> We were able to get the app connected to the MongoDB. We also created User models, and routes to get and add users.
> 
> ### Testing
> How did you confirm your changes worked?
> 
> We tested the routes on postman and were able to get both routes to work properly.
> 
> ### Confirmation of Change
> Upload a screenshot, if possible. Otherwise, please provide instructions on how to see the change.
> 
> Changes can be viewed in the app.ts, models folder, and routes folder.

Add `Resolves #6` so that it auto-closes the issue when PR is merged

YashRavipati1 commented on backend/package.json:
```diff
@@ -14,10 +14,12 @@
   "license": "ISC",
```
Nothing for me to add that Aman didn't address for now. I will review for final approval once the requested changes are made.

adhi0331 commented:
Fixed the ESlint issue and the bug on the get request. Added images below of what the get and post request looks like on my end. 
<img width="1024" alt="Screenshot 2023-01-20 at 12 46 45 PM" src="https://user-images.githubusercontent.com/85322002/213802663-3a42f99d-1cf5-4d14-a223-fd59310bf2cf.png">
<img width="1021" alt="Screenshot 2023-01-20 at 12 46 25 PM" src="https://user-images.githubusercontent.com/85322002/213802685-51de8aaf-4d02-438a-9b54-3c3988c1cbfe.png">

AmanKAggarwal reviewed (approved):
Great work! Approved for merge

YashRavipati1 reviewed (approved):
Everything looks good
